src/boot/boot.janet

```diff
@@ -3736,12 +3736,13 @@
     ~(,ev/thread (fn _spawn-thread [&] ,;body) nil :n))
 
   (defmacro ev/with-deadline
-    `Run a body of code with a deadline, such that if the code does not complete before
-    the deadline is up, it will be canceled.`
-    [deadline & body]
+    ``Run a body of code with a deadline of `sec` seconds, such that if
+    the code does not complete before the deadline is up, it will be
```

Comment:
I'm unsure if this is true, I thought it was `tocancel` that got canceled. The body of code goes into `tocheck`. I'd think that `tocheck` isn't canceled here at all, unless the parent chooses to catch the deadline error and then `cancel` `tocheck`?

Reply:
The default value of `tocancel` is `(fiber/root)`.

Reply:
The following code:

```janet
(let [fib (coro
            (pp [:fib (fiber/current)])
            (pp [:lineage (debug/lineage (fiber/root))])
            (ev/sleep 0.2)
            (yield :hi))]
  (def tocancel (ev/deadline 0.1 nil fib))
  (pp [:tocancel tocancel])
  (resume fib))
```

gives the output (here anyway):

```
(:tocancel <fiber 0x5652813D8EC0>)
(:fib <fiber 0x5652814228A0>)
(:lineage @[<fiber 0x5652813D8EC0> <fiber 0x565281383900> <fiber 0x565281418480> <fiber 0x565281421770> <fiber 0x5652814228A0>])
error: deadline expired
  in ev/sleep [src/core/ev.c] on line 2945
  in <anonymous> [<anonymous>] on line 22, column 13
  in _thunk [<anonymous>] (tailcall) on line 26, column 3
```

Reply:
So the above was meant to be somewhat close to a macroexpansion of `ev/with-deadline`.

For a use of `ev/with-deadline`:

```janet
(ev/with-deadline 0.01
                  (ev/sleep 1)
                  :hi)
```

gives (here anyway):

```
error: deadline expired
  in ev/sleep [src/core/ev.c] on line 2945
  in <anonymous> [repl] on line 2, column 19
  in _thunk [repl] (tailcall) on line 1, column 1
```

I think this is because when `tocancel` is specified as `nil`, the value of `(fiber/root)` is used for `tocancel` instead.

Reply:
The root fiber is trying to resume a fiber constructed by `ev/with-deadline`. If you cancel the root fiber while it's trying to resume a fiber, an error is sent to the fiber through the root fiber.

If the `ev/with-deadline` fiber propagates the error upward, the root fiber is also cancelled.... If the fiber ignores the error, then nothing really happens...

So, it's correct to say that the block of code inside `ev/with-deadline` is cancelled with an error. If you ignore the error, the cancellation stops.

Reply:
I suggest adding something like

> If the body of code inside `ev/with-deadline` discards the error caused by `ev/with-deadline`, `ev/with-deadline` doesn't stop it.